lib/grpc/adapter/cowboy/handler.ex

```diff
@@ -36,7 +36,7 @@ defmodule GRPC.Adapter.Cowboy.Handler do
         compressor: compressor
       }
 
-      pid = spawn_link(__MODULE__, :call_rpc, [server, path, stream])
+      pid = :proc_lib.spawn_link(__MODULE__, :call_rpc, [server, path, stream])
```

Comment:
it didn't quite work as i had hoped, so doing another approach

Reply:
`:proc_lib` makes it a special process http://erlang.org/doc/design_principles/spec_proc.html

part of that includes generating error reports when a process crashes, i.e. https://github.com/erlang/otp/blob/d9516215a82c9297180da0196f0e1e43cfd26535/lib/stdlib/src/proc_lib.erl#L521-L532

And, Bugsnag can listen for those crashes (https://github.com/jarednorman/bugsnag-elixir/blob/525231a514e4627f00112064734942b4b32e4388/lib/bugsnag.ex#L20-L22) and report them.

`:proc_lib.spawn_link` is essentially a wrapper around `spawn_link` that does some extra things to make it a special process.

Generally better to use higher-level modules like `Task`, `GenServer`, `:gen_statem`, etc. instead of `:proc_lib` though.

(I didn't reach for `Task` here because I wanted to avoid doing as much surgery to the module as possible.)

Reply:
Thanks much for the explanation 👍